packages/front-end/components/Settings/WebhooksModal.tsx

```diff
@@ -61,13 +64,29 @@ const WebhooksModal: FC<{
     }
   };
 
+  const isValidHttp = (urlString: string) => {
+    let url;
+    try {
+      url = new URL(urlString);
+    } catch (e) {
+      return false;
+    }
+    return /https?/.test(url.protocol);
+  };
+
   const onSubmit = form.handleSubmit(async (value) => {
+    setDisplayHttpValidationError(false);
     if (value.endpoint.match(/localhost/g)) {
       throw new Error("Invalid endpoint");
     }
+    if (!isValidHttp(value.endpoint)) {
+      setDisplayHttpValidationError(true);
+      return;
```

Comment:
You can just throw from here instead and it will show the message and manage the error state automatically.

Reply:
oh that is nice! good to know